neutronics/mcnp_inputs.py

```diff
@@ -0,0 +1,175 @@
+        the core material.
+        
+        Modified Attributes:
+        --------------------
+            core_vol (float): homogenized core volume
+            refl_vol (float): reflector volume
+            vfrac_cool (float): coolant volume fraction
+            vfrac_cermet (float): cermet matrix volume fraction
+            vfrac_clad (float): cladding volume fraction
+        """
+        # core and reflector volume required for depletion calculation
+        self.core_vol = self.r**2 * math.pi * self.z
+        self.refl_vol = ((self.r + self.refl_t)**2 - self.r**2)*math.pi * self.z
+        
+        pitch = 2*r_cool*PD
+        # calculate 'volumes' for fixed length
+        v_cool = (r_cool ** 2 * math.pi)
+        # clad volume fraction
+        v_clad = ((r_cool + c)**2 - r_cool**2)*math.pi
+        # fuel volume fraction
+        v_cermet = (math.sqrt(3)*pitch**2 / 2.0) - (r_cool + c) ** 2 * math.pi 
+
+        cell_vol = v_cool + v_clad + v_cermet
+        # calculate vfracs from total cell volume
+        self.vfrac_cool = v_cool / cell_vol
+        self.vfrac_clad = v_clad / cell_vol
+        self.vfrac_cermet = v_cermet / cell_vol
+
+        
+    def homog_core(self, enrich=0.9, r_cool=0.5, 
+                         PD=1.48, rho_cool=0.087, c=0.031):
+        """Homogenize the fuel, clad, and coolant.
+        
+        Arguments:
+        ----------
+            enrich (float) (opt): uranium enrichment
+            r_cool (float) (opt): coolant channel radius
+            PD (float) (opt): pitch to diameter ratio
+            rho_cool (float) (opt): coolant density
+            c (float) (opt): claddinng thickness
+        
+        Modified Attributes:
+        --------------------
+            rho (float): fuel density
+            fuel_string (str): MCNP material string 
+        """
+        # get volumes, volume fractions
+        self.calc_vol_vfrac(r_cool, PD, c)
+        
+        # volume-weighted densities/masses
+        fracs = {'fuel' : {
+                      'volfrac' : self.vfrac_cermet*md.vfrac_UN,
+                      'rho' : md.rho_UN},
+                 'matr' : {
+                      'volfrac' : self.vfrac_cermet*(1 - md.vfrac_UN),
+                      'rho' : md.rho_W},
+                 'cool' : {
+                      'volfrac' : self.vfrac_cool,
+                      'rho' : rho_cool}, 
+                 'clad' : {
+                      'volfrac' : self.vfrac_clad,
+                      'rho' : md.rho_In}
+                 }
+
+        # get UN material from custom composition
+        homog_mat = Material(md.enrich_fuel(enrich))
+        homog_mat.mass = fracs['fuel']['volfrac'] * md.rho_UN
+        
+        del fracs['fuel']
+        # load non-fuel materials
+        for mat in fracs:
+            pyne_mat = self.matlib[md.mats[mat]]
+            pyne_mat.mass = fracs[mat]['volfrac']*fracs[mat]['rho']
+            homog_mat += pyne_mat
```

Comment:
I continue to have a conceptual difference of opinion about how this should be done here.  I guess you are probably getting the right answer, but it doesn't make sense to me to change the original material definition and then add it in whole.  

Conceptually, this changes the definition of a material and then adds these custom definitions together.

Conceptually, I prefer to add fractions of canonical material definitions.

The only reason to care about this is for long-term maintainability.